### PR TITLE
Setup WorkingDir for place-tools init container to avoid permission error

### PR DESCRIPTION
**Pull request by barthy1**
# Changes

With 1001 user `run-steps-as-non-root` test can fail because of missing access permissions, see https://github.com/tektoncd/pipeline/issues/3978

Possible solutions are to use 65532 user, which is configured one for gcr.io/distroless/static:nonroot base image or to change working directory to "/" from the default "/home/nonroot" 

"/" working directory allows to use not only 65532 user, so this path is chosen.

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by sbwsg** [comment]
The way this change currently works it appears we're telling users that they may not use any UID in their TaskRun's `securityContext` which is not provided by the distroless image.  But that's an image Tekton is "forcing" on users by having it as the base of the `place-tools` init container.

One of the steps in this example uses uid `2000` but the ubuntu image does not seem to provide this user.  I wonder - does the same error appear when running this step?  I guess maybe not because the ubuntu image has the WORKDIR set to `/`, which is allowed for any UID even if they don't exist in `/etc/passwd`?

I don't think there's anything specific in `place-tools` that requires the workingDir to be `/home/nonroot`.  We could override this in Tekton to be be `/` instead?  That way hopefully a TaskRun's `securityContext` wouldn't trigger this error in our init containers.

File: pkg/pod/entrypoint.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ func orderContainers(entrypointImage string, commonExtraEntrypointArgs []string,
 	initContainer := corev1.Container{
 		Name:  "place-tools",
 		Image: entrypointImage,
+		// Rewrite default WorkingDir from "/home/nonroot" to "/"
+		// as suggested at https://github.com/GoogleContainerTools/distroless/issues/718
```
Comment (sbwsg) [inline pkg/pod/entrypoint.go:99]: 👍 nice reference.

**Review by sbwsg — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**